EPMEDU-736: Create SMS-Code confirmation page

Fileek opened the pull request:
Versions of Jetpack Compose and Kotlin were updated, because in the Compose v1.1.1 was a bug with a numeric keyboard: when a TextField with a keyboardType set to Number get focus, a text keyboard may be displayed for a second or less before numeric keyboard.

MaksimNasalevich committed:
EPMEDU-728
- created sign in screen
- added animeal button

MaksimNasalevich committed:
EPMEDU-728
- fixed detekt issues

MaksimNasalevich committed:
EPMEDU-728
- Added splash screen

MaksimNasalevich committed:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-728_Sign_up_page

 Conflicts:
	app/src/main/java/com/epmedu/animeal/tabs/TabActivity.kt
	settings.gradle.kts

Fileek committed:
[EPMEDU-736]: Add CodeConfirmationScreen.kt implementation

Fileek committed:
[EPMEDU-736]: Add CodeConfirmationScreen.kt implementation

MaksimNasalevich committed:
EPMEDU-728
- Added extensions
- Added navigation module
- moved all resources into resources module

MaksimNasalevich committed:
EPMEDU-728
- Added extensions

MaksimNasalevich committed:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-728_Sign_up_page

 Conflicts:
	app/build.gradle.kts
	app/src/main/java/com/epmedu/animeal/tabs/TabActivity.kt
	base/src/main/java/com/epmedu/animeal/base/theme/Color.kt
	base/src/main/java/com/epmedu/animeal/base/theme/Theme.kt
	feature-more/src/main/java/com/epmedu/animeal/feature/more/MoreScreen.kt
	feature-more/src/main/java/com/epmedu/animeal/feature/more/Screens.kt
	feature/more/src/main/res/values/strings.xml
	library/foundation/src/main/java/com/epmedu/animeal/foundation/topbar/AnimealShortButton.kt
	library/foundation/src/main/java/com/epmedu/animeal/foundation/topbar/TopBar.kt
	settings.gradle.kts

MaksimNasalevich committed:
EPMEDU-728
- Merged conflicts

MaksimNasalevich committed:
EPMEDU-728
- fixed detekt

MaksimNasalevich committed:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-728_Sign_up_page

 Conflicts:
	app/src/main/java/com/epmedu/animeal/tabs/TabActivity.kt
	base/src/main/java/com/epmedu/animeal/base/theme/Color.kt
	base/src/main/java/com/epmedu/animeal/base/theme/Theme.kt

MaksimNasalevich committed:
EPMEDU-728
- fixed conflicts

MaksimNasalevich committed:
EPMEDU-728
- fixed conflicts

MaksimNasalevich committed:
EPMEDU-728
- added dark colors

MaksimNasalevich committed:
EPMEDU-728
- updated github actions

MaksimNasalevich committed:
EPMEDU-728
- moved preview to the bottom

MaksimNasalevich committed:
EPMEDU-728
- fixed issues

MaksimNasalevich committed:
EPMEDU-728
- removed UnusedPrivateMember from previews

Fileek committed:
Remove phone number from parameters of the CodeConfirmationScreen.kt and fix detekt issues

Fileek committed:
Remove phone number from parameters of the CodeConfirmationScreen.kt and fix detekt issues

egorikftp commented:
@Fileek please attach video

Fileek commented:
> @Fileek please attach video

In progress

Fileek commented:
https://user-images.githubusercontent.com/83027107/177181088-491efc1f-7d06-45a0-81c7-b0ce8495e5d6.mp4

egorikftp commented on feature/login/src/main/java/com/epmedu/animeal/feature/login/code/CodeConfirmationScreen.kt:
```diff
@@ -0,0 +1,45 @@
+
+    LaunchedEffect(Unit) {
+        delay(FOCUS_DELAY)
+        focusRequester.requestFocus()
```
Please check if we can remove `delay` by moving `focusRequester.requestFocus()` into `DisposableEffect`

Fileek replied:
> Please check if we can remove `delay` by moving `focusRequester.requestFocus()` into `DisposableEffect`

No, it's crashing, because FocusRequester is not initialized.

egorikftp commented on feature/login/src/main/java/com/epmedu/animeal/feature/login/code/CodeConfirmationScreenUi.kt:
```diff
@@ -0,0 +1,262 @@
+    val seconds = DecimalFormat("00").format(resendDelay)
+    val text =
+        if (isEnabled) stringResource(id = R.string.resend_code)
+        else "${stringResource(id = R.string.resend_code_in)}$seconds"
```
why don't use string resource with args?

Fileek replied:
Done

egorikftp commented on settings.gradle.kts:
```diff
@@ -12,3 +12,4 @@ rootProject.name = "Animeal"
 include(":app")
 include(":base")
 include(":feature:more")
+include(":feature:login")
```
please follow alphabetical order)

Fileek replied:
Done)

egorikftp commented on base/build.gradle.kts:
```diff
@@ -11,4 +11,8 @@ dependencies {
 
     debugImplementation(libs.compose.ui.tooling)
+
+    // Temporary fix for preview bug https://issuetracker.google.com/issues/227767363
```
<img width="406" alt="image" src="https://user-images.githubusercontent.com/16294951/177183581-6603474e-1723-4a9d-a05d-d4745dca1dfe.png">

could you please check on Dolphin?

Fileek replied:
Yes, it's Chipmunk's issue. I've added todo to not forget to remove these dependencies when fix come out.

Fileek committed:
Replace concatenating strings by formatting arguments

Fileek committed:
Replace concatenating strings by formatting arguments

Fileek committed:
Rearrange module dependencies in alphabetical order

Fileek committed:
Add more information about preview bug and add todo

Fileek committed:
Add more information about preview bug and add todo

Fileek committed:
Add private modifier to previews

Fileek committed:
Add private modifier to previews

Fileek committed:
Move composables to separate files

Fileek committed:
Move composables to separate files

MaksimNasalevich committed:
EPMEDU-728
- renamed Spacers, moved shortButton to button package

FlinForm committed:
EPMEDU-754: [Android] Add Firebase Crashlytics to project

Fileek committed:
Fix package declaration

Fileek committed:
Fix package declaration

FlinForm committed:
Fix ci-workflow file

FlinForm committed:
EPMEDU-754: Fix ci-workflow file

MaksimNasalevich committed:
EPMEDU-728
- changed default height

Fileek committed:
Rename colors

Fileek committed:
Rename colors

Fileek committed:
Return no wildcard imports rule in detekt

Fileek committed:
Return no wildcard imports rule in detekt

Fileek committed:
Add dark mode support and previews

Fileek committed:
Add dark mode support and previews

DoTheMonkeyBusiness committed:
EPMEDU-728: SignIn page

EPMEDU-728: SignIn page

- created SignIn screen
- created Splash screen
- added navigation module
- added extensions module
- added foundation module
- added common module
- added resources module
- moved all resources into resources module

FlinForm commented on feature/login/src/main/java/com/epmedu/animeal/feature/login/code/CodeConfirmationScreen.kt:
```diff
@@ -0,0 +1,45 @@
+
+    LaunchedEffect(Unit) {
+        delay(FOCUS_DELAY)
+        focusRequester.requestFocus()
```
Why do we need that delay here?

Fileek replied:
Without delay TextField will not get focus after screen launching

FlinForm commented on feature/login/src/main/java/com/epmedu/animeal/feature/login/code/CodeConfirmationViewModel.kt:
```diff
@@ -0,0 +1,94 @@
+    private fun getEmptyCode() = List(CODE_SIZE) { null }
+
+    private suspend fun launchResendTimer() {
+        for (tick in RESEND_DELAY downTo 1L) {
```
Why do we need this timer here?

Fileek replied:
Do you want to move it in composable or you want to suggest better approach to implement the timer?

FlinForm committed:
EPMEDU-754: Fix PR comments

FlinForm committed:
Merge pull request #24 from AnimealProject/feature/PMEDU-754-Add_Firebase_Crashlytics_to_project

EPMEDU-754: [Android] Add Firebase Crashlytics to project

Fileek committed:
Fix issues after rebase

Fileek committed:
Rename screen and its related components

Fileek committed:
Merge remote-tracking branch 'origin/feature/epmedu-736-create-sms-code-confirmation-screen' into feature/epmedu-736-create-sms-code-confirmation-screen

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/epmedu/animeal/tabs/TabActivity.kt
#	base/src/main/java/com/epmedu/animeal/base/theme/Color.kt
#	feature/login/build.gradle.kts
#	feature/login/src/main/AndroidManifest.xml
#	gradle/libs.versions.toml
#	library/foundation/src/main/java/com/epmedu/animeal/foundation/button/AnimealShortButton.kt
#	settings.gradle.kts

Fileek committed:
Fix issues after rebase
